doc: Add release notes for NCS 1.9 (#6812)
nrfconnect/sdk-nrf: doc/nrf/releases/release-notes-1.9.0.rst

The feature will also be labelled with "EXPERIMENTAL" in Kconfig files to indicate this status.
Build warnings will be generated to indicate when features labelled EXPERIMENTAL are included in builds unless the Kconfig option :kconfig:`CONFIG_WARN_EXPERIMENTAL` is disabled.

Highlights
**********

* Added experimental support for Nordic Distance Toolbox with examples. The software package adds phase-based distance measurement support for the nRF52 and nRF53 Series of devices.
* Added support for Amazon Frustration-Free Setup (FFS) for Matter devices.
* Added deep power-down mode to QSPI driver to optimize energy usage for solutions using an external flash memory.
* Added support for micro:bit v2 board.
* Added support for filtered ephemeris that improves A-GPS download size.
* Updated the support for Periodic Advertising with the SoftDevice Controller. It is no longer experimental.
* Updated the Zigbee solution on nRF5340. It is no longer experimental and is now Zigbee Platform Certification compliant.
* Replaced proprietary sockets with nrf_modem APIs that include link-time optimization, thereby reducing memory usage.

> 

Reviewer: Please add highlight for nrf_security:

* Added experimental support for PSA crypto drivers in nrf_security
* Added PSA crypto drivers for CryptoCell and Oberon

Can be made into one line as well

Author: Moving to the sdk-nrfxlib section as agreed offline.

  * :ref:`event_manager`:

    * Added:

      * ``EVENT_SUBSCRIBE_FIRST`` subscriber priority and updated the documentation about this functionality.

    * Updated:

      * The sections used by the event manager and stopped using orphaned sections.

        * Event manager no longer uses orphaned sections to store information about event types, listeners, and subscribers.    Hence, the :kconfig:`CONFIG_LINKER_ORPHAN_SECTION_PLACE` option is no longer required in the config file.

      * Reworked priorities.

    * Removed:

      *  Forced alignment for x86.

  * :ref:`event_manager_profiler_tracer`:

    * Updated:

      * Event manager Profiler Tracer to no longer use orphaned sections to store profiler information.
        Hence, the :kconfig:`CONFIG_LINKER_ORPHAN_SECTION_PLACE` option is no longer required in the config file.

TF-M libraries
--------------

* Updated:

  * :ref:`lib_tfm_ioctl_api`:

    * Updated the documentation about the IOCTL request types that have been moved to the upstream TF-M platform support.

sdk-nrfxlib
-----------

See the changelog for each library in the :doc:`nfxlib documentation <nrfxlib:README>` for additional information.

* Added:

> 

Reviewer: Referencing point in highlights on nrf_security additions:

Maybe add a point here on adding PSA crypto drivers for CryptoCell and Oberon instead of having that part in the release highlights

Author: Added

    It is now assumed to be 0 dBm during advertising.

  * Documentation and diagrams for the :ref:`nrf_desktop_ble_bond`.
  * Architecture diagrams.
  * Documentation with information about fwupd support.

nRF Pelion client
-----------------

* Removed the nRF Pelion client application as part of removing the Pelion DM support.

nRF9160: Serial LTE modem
-------------------------

* Updated:

  * :ref:`slm_description` as part of adding the overlay for Thingy:91 target:

    * Added:

      * :ref:`slm_testing_twi` in :ref:`slm_testing`.
      * :ref:`slm_connecting_thingy91`.

    * Updated:

      * :ref:`SLM_AT_TWI`.
      * Additional configuration section by adding information about the configuration files for Thingy:91.

  * SLM UART #XSLMUART section in :ref:`SLM_AT_gen`.
  * :ref:`slm_config` with the configuration option related to SLM UART.

Samples
=======

This section provides detailed lists of changes by :ref:`sample <sample>`, including protocol-related samples.
For lists of protocol-specific changes, see `Protocols`_.

Bluetooth samples
-----------------

* Added:

  * :ref:`multiple_adv_sets` sample.
  * :ref:`ble_nrf_dm` sample.

>   * :ref:`bluetooth_direction_finding_central` sample.

Author: Added

  * :ref:`lib_ram_pwrdn` library:

    * Added:

      * Functions for powering up and down RAM sections for a given address range.
      * Experimental functionality to automatically power up and down RAM sections based on the libc heap usage.
      * Support for nRF5340 application core to power up and power down RAM sections.

  * :ref:`event_manager`:

    * Added:

      * ``EVENT_SUBSCRIBE_FIRST`` subscriber priority and updated the documentation about this functionality.

    * Updated:

      * The sections used by the event manager and stopped using orphaned sections.

        * Event manager no longer uses orphaned sections to store information about event types, listeners, and subscribers.    Hence, the :kconfig:`CONFIG_LINKER_ORPHAN_SECTION_PLACE` option is no longer required in the config file.

      * Reworked priorities.

    * Removed:

      *  Forced alignment for x86.

  * :ref:`event_manager_profiler_tracer`:

    * Updated:

      * Event manager Profiler Tracer to no longer use orphaned sections to store profiler information.
        Hence, the :kconfig:`CONFIG_LINKER_ORPHAN_SECTION_PLACE` option is no longer required in the config file.

> TFM libraries

Author: Fixed.

* Replaced proprietary sockets with nrf_modem APIs that include link-time optimization, thereby reducing memory usage.

Sign up for the `nRF Connect SDK v1.9.0 webinar`_ to know more about the new features.

Release tag
***********

The release tag for the |NCS| manifest repository (|ncs_repo|) is **v1.9.0**.
Check the :file:`west.yml` file for the corresponding tags in the project repositories.

To use this release, check out the tag in the manifest repository and run ``west update``.
See :ref:`cloning_the_repositories` for more information.

Supported modem firmware
************************

See `Modem firmware compatibility matrix`_ for an overview of which modem firmware versions have been tested with this version of the |NCS|.

Use the latest version of the nRF Programmer app of `nRF Connect for Desktop`_ to update the modem firmware.
See :ref:`nrf9160_gs_updating_fw_modem` for instructions.

Known issues
************

Known issues are only tracked for the latest official release.
See `known issues for nRF Connect SDK v1.9.0`_ for the list of issues valid for the latest release.

Included projects and versions
******************************

> .. list-table:: Project versions

Author: To be updated.

   :widths: 15 10
   :header-rows: 1

   * - Project name
     - Version
   * - Zephyr
     -
   * - MCUboot
     -
   * - nrfxlib
     -
   * - mbed TLS
     -
   * - TF-M
     -
   * - TF-M Mcuboot
     -
   * - Matter
     -
   * - nrf-802154
     -
   * - Homekit
     -
   * - FindMy
     -
   * - memfault-firmware-sdk
     -
   * - CMock
     -
   * - Unity
     -
   * - Alexa-Gadgets-Embedded-Sample-Code
     -
   * - cddl-gen
     -
   * - CJSON
     -
   * - openthread
     -

> Modem-related libraries and versions

Author: To be updated

  * :ref:`ug_matter_creating_accessory` by updating the described procedure to match the current Matter API.

Thread
------

* Updated:

  * Enabling of Thread 1.2. Refer to :ref:`thread_ug_thread_specification_options` and :ref:`thread_ug_feature_sets` for more information.
  * :ref:`ug_thread_cert` by adding instructions for using VS Code extension as an alternative to command line when building the CLI sample for Thread certification using the Thread Test Harness.
  * :ref:`thread_ug_feature_sets` by adding support for SRP client and removing rarely used features.

Zigbee
------

* Added:

  * Development ZBOSS stack library version based on the ZBOSS build v3.11.1.177+v5.1.1.
    This library version is dedicated for testing ZCL v8 features.
  * ZBOSS libraries variant with ZBOSS Traces enabled.
  * New version of the :ref:`ug_zigbee_tools_ncp_host` (v2.1.1).

* Updated:

  * ZBOSS Zigbee stack to version ``3.11.1.0+5.1.1``.

>     See the :ref:`nrfxlib:zboss_changelog` in the nrfxlib documentation for detailed information.

Reviewer 2: @umapraseeda , the Zigbee team would like to include the release notes listed here:
https://projecttools.nordicsemi.no/confluence/display/KRKNWK/Release+notes+and+known+issues+1.9.0

Basically, the following list:

```
* Added:

  * Experimental support for Zigbee Cluster Library ver8 (ZCL8), included in :ref:`nrfxlib:zboss` v3.11.1.177.
  * Support for Zigbee Base Device Behavior v3.0.1 (BDB 3.0.1).
  * Set of Zigbee libraries with binary trace logs enabled.

* Updated:
  
  * Support for nRF5340 and for the combination of nRF5340 and nRF21540. The support is not experimental anymore.
  * :ref:`nrfxlib:zboss` to v3.11.1.0 and platform v5.1.1 (``v3.11.1.0+v5.1.1``)
  * :ref:`ZBOSS Network Co-processor Host <ug_zigbee_tools_ncp_host>` package to the new version v2.1.1.
  * Zigbee OSIF library with Kconfig options that allow to either reset or halt the device upon ZBOSS stack assert. Reset is enabled by default.

* Fixed several bugs.
* Implemented stability and performance improvements.
* Improved documentation:

  * Added documentation for collecting Zigbee trace logs.
```

Reviewer 2: Uma just asked me whether this new list should replace the whole list in the PR. I think the entry in the lines 167-168 and the mention of the changelog in line 175 are to be used with the new list form Confluence, as they are more detailed.

@maciejpietras , any opinion? The list needs to be reviewed and approved by someone from the development team, so I cannot take this call on my own.

Reviewer 3: Please go on with Confluence version as reviewed by team and PMT.

For lists of protocol-specific changes, see `Protocols`_.

Bluetooth samples
-----------------

* Added:

  * :ref:`multiple_adv_sets` sample.
  * :ref:`ble_nrf_dm` sample.

* Updated:

  * :ref:`peripheral_rscs` sample:

    * Corrected the number of bytes for setting the Total Distance Value and specified the data units. See :ref:`peripheral_rscs_testing` for more details.

  * :ref:`direct_test_mode` sample:

    * Added support for front-end module devices that support 2-pin PA/LNA interface with additional support for the Skyworks SKY66114-11 and the Skyworks SKY66403-11. See the Skyworks front-end module and Vendor-specific packet payload sections in the sample documentation.

  * :ref:`peripheral_hids_mouse` and :ref:`peripheral_hids_keyboard` samples:

    * Added a notice about encryption requirement.

  * :ref:`central_and_peripheral_hrs` sample:

    * Clarified the Requirements section and fixed a link in the Testing section of the sample documentation.

  * :ref:`peripheral_uart` sample:

    * Increased the system workqueue stack size for the :file:`prj_minimal.conf` file.

> 

Reviewer 4: Please add a following statement (for https://github.com/nrfconnect/sdk-nrf/pull/6794):
```suggestion
* Aligned all samples to use the Bluetooth helper macros: :c:macro:`BT_UUID_16_ENCODE` and :c:macro:`BT_CONN_CB_DEFINE`.
```

Author: Added

Unity
-----

* Fixed a bug that resulted in the mocks for some functions not having the ``__wrap_`` prefix. This happened for functions declared with whitespaces between identifier and parameter list.

HID Configurator for nRF Desktop
--------------------------------

* Updated:

  * The script to recognize the bootloader variant as a DFU module variant for the configuration channel communication. The new implementation is backward compatible - the new version of the script checks for module name and acts accordingly. See the Performing DFU section in :ref:`nrf_desktop_config_channel_script` documentation.

  * Documentation on setting up the dependencies for Debian/Ubuntu/Linux Mint operating system.

MCUboot
=======

The MCUboot fork in |NCS| (``sdk-mcuboot``) contains all commits from the upstream MCUboot repository up to and including ``680ed07``, plus some |NCS| specific additions.

The code for integrating MCUboot into |NCS| is located in the :file:`ncs/nrf/modules/mcuboot` folder.

The following list summarizes both the main changes inherited from upstream MCUboot and the main changes applied to the |NCS| specific additions:

* |no_changes_yet_note|

> Mcumgr

Author: @nvlsianpu and @de-nordic , please check this section.

Reviewer 5: * mcumgr module has been meld into the zephyr-rtos codebase

Reviewer 6: Please delete the whole mcumgr section

======

The mcumgr library contains all commits from the upstream mcumgr repository up to and including snapshot ``657deb65``.

The following list summarizes the most important changes inherited from upstream mcumgr:

* |no_changes_yet_note|

Zephyr
======

.. NOTE TO MAINTAINERS: All the Zephyr commits in the below git commands must be handled specially after each upmerge and each NCS release.

The Zephyr fork in |NCS| (``sdk-zephyr``) contains all commits from the upstream Zephyr repository up to and including ``3f82656``, plus some |NCS| specific additions.

For a complete list of upstream Zephyr commits incorporated into |NCS| since the most recent release, run the following command from the :file:`ncs/zephyr` repository (after running ``west update``):

.. code-block:: none

   git log --oneline 3f82656 ^v2.6.0-rc1-ncs1

For a complete list of |NCS| specific commits, run:

.. code-block:: none

>    git log --oneline manifest-rev ^3f82656

Author: @nvlsianpu, please double-check.

The current |NCS| main branch is based on the Zephyr v2.7 development branch.

Matter (Project CHIP)
=====================

> The Matter fork in the |NCS| (``sdk-connectedhomeip``) contains all commits from the upstream Matter repository up to, and including, ``77ab003e5fcd409cd225b68daa3cdaf506ed1107``.

Author: @Damian-Nordic , please check the commit hash.

Reviewer 7: Seems correct.

The feature is made available in its current state though the design and interfaces can change between release tags.
The feature will also be labelled with "EXPERIMENTAL" in Kconfig files to indicate this status.
Build warnings will be generated to indicate when features labelled EXPERIMENTAL are included in builds unless the Kconfig option :kconfig:`CONFIG_WARN_EXPERIMENTAL` is disabled.

Highlights
**********

* Added experimental support for Nordic Distance Toolbox with examples. The software package adds phase-based distance measurement support for the nRF52 and nRF53 Series of devices.
* Added support for Amazon Frustration-Free Setup (FFS) for Matter devices.
* Added deep power-down mode to QSPI driver to optimize energy usage for solutions using an external flash memory.
* Added support for micro:bit v2 board.
* Added support for filtered ephemeris that improves A-GPS download size.
* Updated the support for Periodic Advertising with the SoftDevice Controller. It is no longer experimental.

> * Updated the Zigbee solution on nRF5340. It is no longer experimental and is now Zigbee Platform Certification compliant.

Reviewer 3: Can we claim it before having certification in hands?

Reviewer 2: I think this is an unfortunately phrased line, given that "compliant" can be interpreted as both "meeting the certification requirements" (thus suggesting the certification is granted) and "ready for certification".
Maybe this would be a better phrasing?
```suggestion
* Updated the Zigbee solution on nRF5340. It is no longer experimental and is now compliant with Zigbee Platform Certification requirements.
```
or even:
```suggestion
* Updated the Zigbee solution on nRF5340. It is no longer experimental and is now compliant with Zigbee Platform Certification requirements. The official certification is pending.
```

Author: Retaining the original statement as suggested by Krzysztof Loska.

* Added support for Amazon Frustration-Free Setup (FFS) for Matter devices.
* Added deep power-down mode to QSPI driver to optimize energy usage for solutions using an external flash memory.
* Added support for micro:bit v2 board.
* Added support for filtered ephemeris that improves A-GPS download size.
* Updated the support for Periodic Advertising with the SoftDevice Controller. It is no longer experimental.
* Updated the Zigbee solution on nRF5340. It is no longer experimental and is now Zigbee Platform Certification compliant.
* Replaced proprietary sockets with nrf_modem APIs that include link-time optimization, thereby reducing memory usage.

Sign up for the `nRF Connect SDK v1.9.0 webinar`_ to know more about the new features.

Release tag
***********

The release tag for the |NCS| manifest repository (|ncs_repo|) is **v1.9.0**.
Check the :file:`west.yml` file for the corresponding tags in the project repositories.

To use this release, check out the tag in the manifest repository and run ``west update``.
See :ref:`cloning_the_repositories` for more information.

For information on the included repositories and revisions, see `Repositories and revisions`_.

> 

Reviewer 8: Due to the pending deprecation of SES, I think we need a notice in the NCS v1.9.  Suggest:

IDE support
*************
nRF Connect SDK version 1.9.0 is supported by both [nRF Connect extensions for Visual Studio Code](https://nrfconnect.github.io/vscode-nrf-connect/) and Segger Embedded Studio Nordic Edition.  In future releases, documentation on the setup and usage of Segger Embedded Studio Nordic Edition will no longer be available and all references to IDE support will instruct users with nRF Connect extensions for Visual Studio Code.  In future releases, Segger Embedded Studio will no longer be tested with the build system and it is not recommended for new projects.

nRF Connect extensions for Visual Studio Code will be under continued active development and supported and tested with future nRF Connect SDK releases. It is recommended for all new projects.

Author: @jdstapes, thanks for the suggestion! I have added this section.

      * Reworked priorities.

    * Removed:

      *  Forced alignment for x86.

  * :ref:`event_manager_profiler_tracer`:

    * Updated:

      * Event manager Profiler Tracer to no longer use orphaned sections to store profiler information.
        Hence, the :kconfig:`CONFIG_LINKER_ORPHAN_SECTION_PLACE` option is no longer required in the config file.

TF-M libraries
--------------

* Updated:

  * :ref:`lib_tfm_ioctl_api`:

    * Updated the documentation about the IOCTL request types that have been moved to the upstream TF-M platform support.

sdk-nrfxlib
-----------

See the changelog for each library in the :doc:`nfxlib documentation <nrfxlib:README>` for additional information.

* Added:

  * :ref:`nrfxlib:nrf_dm` and corresponding overview, usage, changelog, and API documentation.

* Updated:

  * :ref:`crypto`:

    * Added PSA crypto drivers for CryptoCell and Oberon.
    * Updated:

      * :ref:`nrfxlib:nrf_cc3xx_platform_readme` to v0.9.13.
      * :ref:`nrfxlib:nrf_cc3xx_mbedcrypto_readme` to v0.9.13.
      * :ref:`nrfxlib:nrf_oberon_readme` to v3.0.10.

>     * Deprecated the ability of Mbed TLS API to enable multiple cryptographic backends in a build, such as the Oberon accelerated library or CryptoCell hardware accelerator. This applies only to the APIs with the ``mbedtls_`` prefix.

Author: added.

    * Updated:

      * :ref:`nrfxlib:nrf_modem` to v1.5.1.
      * Architecture diagram in :ref:`nrfxlib:architecture`.
      * :ref:`nrfxlib:nrf_modem_delta_dfu`.

    * Removed:

      * Overview diagram in :ref:`nrfxlib:nrf_modem`.
      * AT socket page.
      * Extensions page.

  * :ref:`nrf_security`:

    * Added experimental support for PSA crypto drivers in :ref:`nrf_security`.

Scripts
=======

This section provides detailed lists of changes by :ref:`script <scripts>`.

Unity
-----

* Fixed a bug that resulted in the mocks for some functions not having the ``__wrap_`` prefix. This happened for functions declared with whitespaces between identifier and parameter list.

HID Configurator for nRF Desktop
--------------------------------

* Updated:

  * The script to recognize the bootloader variant as a DFU module variant for the configuration channel communication. The new implementation is backward compatible - the new version of the script checks for module name and acts accordingly. See the Performing DFU section in :ref:`nrf_desktop_config_channel_script` documentation.

  * Documentation on setting up the dependencies for Debian/Ubuntu/Linux Mint operating system.

> MCUboot

Author: @nvlsianpu please check the section.

=======

The MCUboot fork in |NCS| (``sdk-mcuboot``) contains all commits from the upstream MCUboot repository up to and including ``680ed07``, plus some |NCS| specific additions.

The code for integrating MCUboot into |NCS| is located in the :file:`ncs/nrf/modules/mcuboot` folder.

The following list summarizes both the main changes inherited from upstream MCUboot and the main changes applied to the |NCS| specific additions:

* |no_changes_yet_note|

Mcumgr
======

The mcumgr library contains all commits from the upstream mcumgr repository up to and including snapshot ``657deb65``.

The following list summarizes the most important changes inherited from upstream mcumgr:

* |no_changes_yet_note|

> Zephyr

Author: @carlescufi , could you check this section?

Reviewer 6: @SebastianBoe  and @mbolivar-nordic please go through this section.

Reviewer 9: done

The Zephyr fork in |NCS| (``sdk-zephyr``) contains all commits from the upstream Zephyr repository up to and including ``3f82656``, plus some |NCS| specific additions.

For a complete list of upstream Zephyr commits incorporated into |NCS| since the most recent release, run the following command from the :file:`ncs/zephyr` repository (after running ``west update``):

.. code-block:: none

   git log --oneline 3f82656 ^v2.6.0-rc1-ncs1

For a complete list of |NCS| specific commits, run:

.. code-block:: none

   git log --oneline manifest-rev ^3f82656

The current |NCS| main branch is based on the Zephyr v2.7 development branch.

Matter (Project CHIP)
=====================

The Matter fork in the |NCS| (``sdk-connectedhomeip``) contains all commits from the upstream Matter repository up to, and including, ``77ab003e5fcd409cd225b68daa3cdaf506ed1107``.

The following list summarizes the most important changes inherited from the upstream Matter:

* Added:

  * Support for General Diagnostics, Software Diagnostics and Thread Network Diagnostics clusters.
  * Initial support for the Matter OTA (Over-the-air) Requestor role, used for updating the device firmware.

> cddl-gen

Author: @oyvindronningstad, could you check this section?

    It is now assumed to be 0 dBm during advertising.

  * Documentation and diagrams for the :ref:`nrf_desktop_ble_bond`.
  * Architecture diagrams.
  * Documentation with information about fwupd support.

nRF Pelion client
-----------------

* Removed the nRF Pelion client application as part of removing the Pelion DM support.

nRF9160: Serial LTE modem
-------------------------

* Updated:

  * :ref:`slm_description` as part of adding the overlay for Thingy:91 target:

    * Added:

      * :ref:`slm_testing_twi` in :ref:`slm_testing`.
      * :ref:`slm_connecting_thingy91`.

    * Updated:

      * :ref:`SLM_AT_TWI`.
      * Additional configuration section by adding information about the configuration files for Thingy:91.

  * SLM UART #XSLMUART section in :ref:`SLM_AT_gen`.
  * :ref:`slm_config` with the configuration option related to SLM UART.

Samples
=======

This section provides detailed lists of changes by :ref:`sample <sample>`, including protocol-related samples.
For lists of protocol-specific changes, see `Protocols`_.

Bluetooth samples
-----------------

* Added:

  * :ref:`multiple_adv_sets` sample.

>   * :ref:`ble_nrf_dm` sample.

Reviewer 4: Can we add a note it uses the new DDFS GATT service? For example:
 ```suggestion
   * :ref:`ble_nrf_dm` sample which uses the new GATT library: :ref:ddfs_readme.
```

Author: Added

      * :ref:`nrfxlib:nrf_modem` to v1.5.1.
      * Architecture diagram in :ref:`nrfxlib:architecture`.
      * :ref:`nrfxlib:nrf_modem_delta_dfu`.

    * Removed:

      * Overview diagram in :ref:`nrfxlib:nrf_modem`.
      * AT socket page.
      * Extensions page.

  * :ref:`nrf_security`:

    * Added experimental support for PSA crypto drivers in :ref:`nrf_security`.

Scripts
=======

This section provides detailed lists of changes by :ref:`script <scripts>`.

Unity
-----

* Fixed a bug that resulted in the mocks for some functions not having the ``__wrap_`` prefix. This happened for functions declared with whitespaces between identifier and parameter list.

HID Configurator for nRF Desktop
--------------------------------

* Updated:

  * The script to recognize the bootloader variant as a DFU module variant for the configuration channel communication. The new implementation is backward compatible - the new version of the script checks for module name and acts accordingly. See the Performing DFU section in :ref:`nrf_desktop_config_channel_script` documentation.

  * Documentation on setting up the dependencies for Debian/Ubuntu/Linux Mint operating system.

MCUboot
=======

> The MCUboot fork in |NCS| (``sdk-mcuboot``) contains all commits from the upstream MCUboot repository up to and including ``680ed07``, plus some |NCS| specific additions.

Author: Updated

The code for integrating MCUboot into |NCS| is located in the :file:`ncs/nrf/modules/mcuboot` folder.

The following list summarizes both the main changes inherited from upstream MCUboot and the main changes applied to the |NCS| specific additions:

> 

Reviewer 5: * Fix support for single application with serial recovery

> 

Author: Added

  * :ref:`thread_ug_feature_sets` by adding support for SRP client and removing rarely used features.

Zigbee
------

* Added:

  * Experimental support for Zigbee Cluster Library ver8 (ZCL8), included in :ref:`nrfxlib:zboss` v3.11.1.177.
  * Support for Zigbee Base Device Behavior v3.0.1 (BDB 3.0.1).
  * Set of Zigbee libraries with binary trace logs enabled.
  * Stability and performance improvements.
  * Documentation for collecting Zigbee trace logs. See :ref:`ug_zigbee_configuring_zboss_traces` for more information.

* Updated:

  * Support for nRF5340 and for the combination of nRF5340 and nRF21540. The support is not experimental anymore.
  * :ref:`nrfxlib:zboss` to v3.11.1.0 and platform v5.1.1 (``v3.11.1.0+v5.1.1``).
  * :ref:`ZBOSS Network Co-processor Host <ug_zigbee_tools_ncp_host>` package to the new version v2.1.1.
  * :ref:`lib_zigbee_osif` library with Kconfig options that allow to either reset or halt the device upon ZBOSS stack assert. Reset is enabled by default.

* Fixed:

  * Several bugs.

HomeKit
-------

* Added:

>   * Nordic proprietary Device Firmware Update over Bluetooth LE (Nordic DFU) based on mcumgr.

Reviewer 10: ```suggestion
  * Nordic proprietary Device Firmware Update over Bluetooth LE (Nordic DFU) based on mcumgr.
    Before using the new DFU solution, make sure to check HomeKit known issues for this release.
```
@umapraseeda We will be a adding a known issue for this and it would be nice to ref to it from here.
@maciejpietras  , could you provide text for the new known issue?

Reviewer 3: We will update known issue list until tomorrow.

Author: Added this statement.

Author: Removing this statement as suggested by @maciejpietras .

>   * Development support for :ref:`matter_weather_station_app` application for HAP over Thread.

Reviewer 11: @ktaborowski should it link to the Matter weather station?

Reviewer 12: No, it's a typo. Thanks for catching

Reviewer 10: This should not be a ref at all , just a name of the app. We cannot link to HomeKit docs from here.

Author: Fixed this.

The following list summarizes both the main changes inherited from upstream MCUboot and the main changes applied to the |NCS| specific additions:

* Added:

  * Check of reset address in the incoming image validation phase. See :kconfig:`CONFIG_MCUBOOT_VERIFY_IMG_ADDRESS`.

* Updated:

  * Always call :c:func:`sys_clock_disable` in main since the empty :c:func:`sys_clock_disable` callback is provided if the platform does not support system clock disable capability.
  * Bootutil library to close flash_area after failure to read swap status information.
  * Allow image header bigger than 1 KB for encrypted images.
  * Zephyr:RSA to use a smaller SHA256 implementation.

* Fixed:

  * Zephyr/boot_serial_extension build failure on :c:macro:`MCUBOOT_LOG_MODULE` when LOG is disabled.
  * Status offset calculation in case of scratch area.
  * Build with image encryption using RSA (issue introduced by migration to MbedTLS 3.0.0).
  * Support for single application with serial recovery.

Zephyr
======

.. NOTE TO MAINTAINERS: All the Zephyr commits in the below git commands must be handled specially after each upmerge and each NCS release.

> The Zephyr fork in |NCS| (``sdk-zephyr``) contains all commits from the upstream Zephyr repository up to and including ``3f82656``, plus some |NCS| specific additions.

Reviewer 9: ```suggestion
The Zephyr fork in |NCS| (``sdk-zephyr``) contains all commits from the upstream Zephyr repository up to and including ``45ef0d2``, plus some |NCS| specific additions.
```

Author: Updated

> For a complete list of upstream Zephyr commits incorporated into |NCS| since the most recent release, run the following command from the :file:`ncs/zephyr` repository (after running ``west update``):

Author: Updated

.. code-block:: none

>    git log --oneline 3f82656 ^v2.6.0-rc1-ncs1

Reviewer 9: Due to the high number of fromtree commits, I don't think this list is particularly useful.

It can't be used to answer any interesting questions that a customer might have that I can think of.

E.g. to answer "Is zephyr commit X included?" or "Which commits are included?"
you really need to look at the fromlist commits as well.

I suggest we replace it with a command that also prints out our own patches.

Reviewer 9: oh, I guess in tandem with the below command it is OK

>    git log --oneline 3f82656 ^v2.6.0-rc1-ncs1

Reviewer 9: ```suggestion
   git log --oneline 45ef0d2 ^zephyr-v2.7.0
```

Reviewer 9: tested to be OK

Author: Updated

> For a complete list of |NCS| specific commits, run:

Author: Updated

.. code-block:: none

>    git log --oneline manifest-rev ^3f82656

Reviewer 9: ```suggestion
   git log --oneline manifest-rev ^45ef0d2
```

Tested to be OK

Author: Updated

> The current |NCS| main branch is based on the Zephyr v2.7 development branch.

Reviewer 9: ```suggestion
The current |NCS| main branch is based on revision 45ef0d2 of Zephyr, which is located between v2.7.0 and v3.0.0 of Zephyr.
```

Author: Updated

* Updated:

  * Always call :c:func:`sys_clock_disable` in main since the empty :c:func:`sys_clock_disable` callback is provided if the platform does not support system clock disable capability.
  * Bootutil library to close flash_area after failure to read swap status information.
  * Allow image header bigger than 1 KB for encrypted images.
  * Zephyr:RSA to use a smaller SHA256 implementation.

* Fixed:

  * Zephyr/boot_serial_extension build failure on :c:macro:`MCUBOOT_LOG_MODULE` when LOG is disabled.
  * Status offset calculation in case of scratch area.
  * Build with image encryption using RSA (issue introduced by migration to MbedTLS 3.0.0).
  * Support for single application with serial recovery.

Mcumgr
======

The mcumgr library contains all commits from the upstream mcumgr repository up to and including snapshot ``657deb65``.

The following list summarizes the most important changes inherited from upstream mcumgr:

* |no_changes_yet_note|

Zephyr
======

.. NOTE TO MAINTAINERS: All the Zephyr commits in the below git commands must be handled specially after each upmerge and each NCS release.

> The Zephyr fork in |NCS| (``sdk-zephyr``) contains all commits from the upstream Zephyr repository up to and including ``3f82656``, plus some |NCS| specific additions.

Reviewer 6: @SebastianBoe and please provide a commit here.

Reviewer 5: 45ef0d2aef9bbba92460e19129b35f0da9729610 is the correct one

Reviewer 9: I can confirm it is 45ef

For a complete list of upstream Zephyr commits incorporated into |NCS| since the most recent release, run the following command from the :file:`ncs/zephyr` repository (after running ``west update``):

.. code-block:: none

   git log --oneline 3f82656 ^v2.6.0-rc1-ncs1

For a complete list of |NCS| specific commits, run:

.. code-block:: none

   git log --oneline manifest-rev ^3f82656

> The current |NCS| main branch is based on the Zephyr v2.7 development branch.

Reviewer 5: That's not true anymore @SebastianBoe Please confirm.

Reviewer 9: suggested a more precise statement.
